components/team_general_tab/open_invite.tsx

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useState} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import type {ActionResult} from 'mattermost-redux/types/actions';
+import type {Team} from 'mattermost-redux/types/teams';
+
+import SettingItemMax from 'components/setting_item_max.jsx';
+import SettingItemMin from 'components/setting_item_min';
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+
+type Props = {
+    teamId: string;
+    isActive: boolean;
+    isGroupConstrained?: boolean;
+    allowOpenInvite?: boolean;
+    onToggle: (active: boolean) => void;
+    patchTeam: (patch: Partial<Team>) => Promise<ActionResult>;
+};
+
+const OpenInvite = (props: Props) => {
+    const intl = useIntl();
+    const [serverError, setServerError] = useState('');
```

Comment:
nit: 1/5 I would prefer to destructure the relevant properties from `props`, but I am not certain how this is being handled over the codebase in general ... I tend to use my personal rule-of-thumb: "if you write `props.x` or `this.props.x` it is best to destructure `x` from it"

Reply:
fair enough, I'll do the changes.

Reply:
i just noticed i forgot an important piece from my rule-of-thumb: "if you write `props.x` or `this.props.x` more than twice it is best to destructure `x` from it"

Sorry for missing that while writing the comment! 🙇🏼‍♂️